### ISSUE TITLE
ALX: a function that prints strings, followed by a new line (variadic function)

### ISSUE THREAD

**Issue by ahmed149** [ahmed149/learning-programing-in-C#74]
#include "variadic_functions.h"
#include <stdarg.h>
#include <stdio.h>

/**
 * print_strings - function to print strings
 * @separator: string separator
 * @n: number of arguments
 * Return: void
 */
void print_strings(const char *separator, const unsigned int n, ...)
{
	va_list list;
	unsigned int a;
	char *str;

	va_start(list, n);
	for (a = 0; a < n; a++)
	{
		str = va_arg(list, char *);
		if (str == NULL)
			printf("(nil)");
		else
			printf("%s", str);
		if ((a != n - 1) && separator != NULL)
			printf("%s", separator);
	}
	printf("\n");
	va_end(list);
}

**Comment by ahmed149** [comment]
This C function, `print_strings`, is a variadic function that prints a variable number of strings to the standard output, with a specified separator string between them. Here's an explanation of each part of the function:

1. `#include "variadic_functions.h"` and `#include <stdarg.h>`:
   - These lines include the user-defined header file "variadic_functions.h" and the standard C library `<stdarg.h>`. The latter is essential for working with variable-length argument lists.

2. `#include <stdio.h>`:
   - This line includes the standard C library `<stdio.h>` for input and output operations, as the function uses `printf` to print the strings.

3. `void print_strings(const char *separator, const unsigned int n, ...)`:
   - This is the function declaration for `print_strings`. It's defined to take a variable number of arguments represented by `...`. It also takes two additional parameters:
     - `separator`: A string used to separate the printed strings.
     - `n`: The number of expected string arguments.

4. `va_list list;`:
   - This line declares a `va_list` named `list`, which will be used to manage the variable-length argument list.

5. `unsigned int a;`:
   - This line declares an unsigned integer variable `a`, which will be used as a loop counter.

6. `char *str;`:
   - This line declares a character pointer `str`, which will be used to store each string argument.

7. `va_start(list, n);`:
   - This line initializes the `list` variable to point to the first unnamed argument after `n`. In other words, it prepares to traverse the variable-length argument list.

8. `for (a = 0; a < n; a++)`:
   - This loop iterates `n` times, where `n` is the number of expected arguments.

9. `str = va_arg(list, char *);`:
   - Inside the loop, it uses `va_arg` to retrieve the next argument from the variable-length list (`list`) as a character pointer (string).

10. `if (str == NULL)`:
    - This condition checks whether the current string argument is NULL.

11. `printf("(nil)");`:
    - If the condition is true (the string is NULL), it prints "(nil)" to indicate that the string is empty or undefined.

12. `else`:
    - If the string is not NULL, it proceeds to print the string using `printf("%s", str)`.

13. `if ((a != n - 1) && separator != NULL)`:
    - This condition checks whether the current iteration is not the last one (`a != n - 1`) and whether a non-null separator string is provided (`separator != NULL`).

14. `printf("%s", separator);`:
    - If the condition is met, it prints the separator string using `printf`.

15. `printf("\n");`:
    - After processing all the strings and separators, it prints a newline character to end the line of output.

16. `va_end(list);`:
    - Finally, this line cleans up the `list` variable, signaling that you've finished working with the variable-length argument list.

In summary, this function allows you to pass a variable number of string arguments to it, along with a separator string. It then prints the strings with the specified separator between them, handling NULL strings by printing "(nil)". It uses `va_start`, `va_arg`, and `va_end` to access the variable-length argument list safely and checks for the presence of a separator string.
